### PR TITLE
Utils port to 3.4 keeping py2 compatibility

### PR DESCRIPTION
**Pull request by scone**
Fixed byte string formatting for hexdump.

= test session starts =
platform darwin -- Python 3.4.1, pytest-2.8.2, py-1.4.30, pluggy-0.3.1
rootdir: /Users/samc/src/python/netlib, inifile:
collected 11 items

netlib/test/test_utils.py ...........

= 11 passed in 0.23 seconds =

### CODE REVIEW

File: test/test_wsgi.py [hunk 1]
```diff
@@ -12,8 +12,6 @@ def tflow():
 
 class TestApp:
 
-    def __init__(self):
-        self.called = False
```
Comment (mhils) [inline test/test_wsgi.py]: Why this change? :smiley:

**Comment by mhils** [comment]
Good stuff, thanks. Once my comment above is resolved, this is ready to merge :-)

File: test/test_wsgi.py (hunk 1 above)
Comment (scone) [inline test/test_wsgi.py]: Py.test gave a warning that it could not collect TestApp because there was an __init__ constructor. 

Maybe this warning isn't an issue for the tests. I'm not familiar with pytest. I decided to err on the side of no warnings. 

Thanks,
Sam

Comment (mhils) [inline test/test_wsgi.py]: py.test shouldn't collect TestApp at all, as TestApp does not contain tests but is an example app used for testing. Thanks for catching that. Can you please just rename TestApp to ExampleApp or DummyApp and re-add the constructor?

**Comment by mhils** [comment]
Thanks! Sorry for not getting back earlier, but you don't get a GitHub notification if new commits are pushed to a PR. Please feel free to add a quick comment next time. :smiley:
